patches/server/0010-Adventure.patch

```diff
@@ -2605,6 +2605,27 @@ index 5a541e14caccaca97759879fd081becd0e8c6af2..13859822feebb65f14973aa688a54fcb
          buf.writeComponent(this.header);
          buf.writeComponent(this.footer);
      }
+diff --git a/src/main/java/net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java b/src/main/java/net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java
+index a5578cbce2d1d39ba3315e53425d5323e6823a65..2b1d863838caa200475d3b8ac694cab7b7454c72 100644
+--- a/src/main/java/net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java
++++ b/src/main/java/net/minecraft/network/protocol/login/ClientboundLoginDisconnectPacket.java
+@@ -17,7 +17,15 @@ public class ClientboundLoginDisconnectPacket implements Packet<ClientLoginPacke
+ 
+     @Override
+     public void write(FriendlyByteBuf buf) {
+-        buf.writeUtf(Component.Serializer.toJson(this.reason));
++        // Paper start - Adventure
++        if (this.reason instanceof io.papermc.paper.adventure.AdventureComponent adv) {
++            // In the login phase, buf.adventure$locale field is always null
++            buf.writeUtf(io.papermc.paper.adventure.PaperAdventure.asJsonString(adv.adventure$component(), null), FriendlyByteBuf.MAX_COMPONENT_STRING_LENGTH);
++            return;
++        }
++        // Paper end - Adventure
++
++        buf.writeUtf(Component.Serializer.toJson(this.reason), FriendlyByteBuf.MAX_COMPONENT_STRING_LENGTH); // Paper - Adventure (Fix vanilla serialized component length)
```

Comment:
This also needs to render non-adventure components. We started doing that elsewhere and I think its probably good to cover all the bases. 

See ComponentSerialization#LOCALIZED_CODECS and just use the Locale.US one from there.